envparse.py

```diff
@@ -1,197 +1,217 @@
-    def test_float(self):
-        self.assertTypeAndValue(float, 33.3, env('FLOAT_VAR', cast=float))
-
-    def test_unicode(self):
-        self.assertTypeAndValue(unicode, u'ubar', env('UNICODE_VAR',
-                                cast=unicode))
-
-    def test_bool_true(self):
-        self.assertTypeAndValue(bool, True, env('BOOL_TRUE_VAR', cast=bool))
-
-    def test_bool_false(self):
-        self.assertTypeAndValue(bool, False, env('BOOL_FALSE_VAR', cast=bool))
-
-    def test_old_style_proxied_value(self):
-        self.assertTypeAndValue(str, 'bar', env('OLD_STYLE_PROXIED_VAR'))
-
-    def test_new_style_proxied_value(self):
-        self.assertTypeAndValue(str, 'bar', env('NEW_STYLE_PROXIED_VAR'))
-
-    def test_int_list(self):
-        self.assertTypeAndValue(list, [42, 33], env('INT_LIST', cast=[int]))
-
-    def test_str_list_with_spaces(self):
-        self.assertTypeAndValue(list, [' foo', '  bar'],
-                                env('STR_LIST_WITH_SPACES', cast=[str]))
-
-    def test_empty_list(self):
-        self.assertTypeAndValue(list, [], env('EMPTY_LIST', cast=[int]))
-
-    def test_schema(self):
-        env = Env(INT_VAR=int, NOT_PRESENT_VAR=(float, 33.3), STR_VAR=str,
-                  INT_LIST=[int], DEFAULT_LIST=([int], [2]))
-
-        self.assertTypeAndValue(int, 42, env('INT_VAR'))
-        self.assertTypeAndValue(float, 33.3, env('NOT_PRESENT_VAR'))
-
-        self.assertTypeAndValue(str, 'bar', env('STR_VAR'))
-        self.assertTypeAndValue(str, 'foo', env('NOT_PRESENT2', default='foo'))
-
-        self.assertTypeAndValue(list, [42, 33], env('INT_LIST'))
-        self.assertTypeAndValue(list, [2], env('DEFAULT_LIST'))
-
-        # Override schema in this one case
-        self.assertTypeAndValue(str, '42', env('INT_VAR', cast=str))
-
-
-if __name__ == "__main__":
-    unittest.main()
+                if cast is None:
+                    cast = params
+        # Default cast is `str` if it is not specified. Most types will be
+        # implicitly strings so reduces having to specify.
+        cast = str if cast is None else cast
+
+        try:
+            value = os.environ[var]
+        except KeyError:
+            if default is NOTSET:
+                error_msg = "Environment variable '{}' not set.".format(var)
+                raise ConfigurationError(error_msg)
+            else:
+                value = default
+
+        # Resolve any proxied values
+        if hasattr(value, 'startswith') and value.startswith('{{'):
+            value = self.__call__(value.lstrip('{{}}'), default, cast, subcast,
+                                  default, force, preprocessor, postprocessor)
+
+        if preprocessor:
+            value = preprocessor(value)
+        if value != default or force:
+            value = self.cast(value, cast, subcast)
+        if postprocessor:
+            value = postprocessor(value)
+        return value
+
+    @classmethod
+    def cast(cls, value, cast=str, subcast=None):
```

Comment:
I think this should be `_cast` because it's not part of the public API.

Again, this is minor, and we can probably merge w/o the fix.